checks/image/image_check.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2019 DigitalOcean
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package image
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/digitalocean/clusterlint/checks"
+	"github.com/digitalocean/clusterlint/kube"
+	"github.com/docker/distribution/reference"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func init() {
+	checks.Register(&containerImageCheck{})
+}
+
+type containerImageCheck struct{}
+
+// Name returns a unique name for this check.
+func (pc *containerImageCheck) Name() string {
+	return "container-images"
+}
+
+// Groups returns a list of group names this check should be part of.
+func (pc *containerImageCheck) Groups() []string {
+	return []string{"image"}
+}
+
+// Description returns a detailed human-readable description of what this check
+// does.
+func (pc *containerImageCheck) Description() string {
+	return "Checks if there are pods with containers in privileged mode"
+}
+
+// Run runs this check on a set of Kubernetes objects. It can return warnings
+// (low-priority problems) and errors (high-priority problems) as well as an
+// error value indicating that the check failed to run.
+func (pc *containerImageCheck) Run(ctx context.Context, objects *kube.Objects) ([]checks.Diagnostic, error) {
+	var diagnostics []checks.Diagnostic
+	repos := ctx.Value(checks.Repositories)
+	validRepos := make(map[string]struct{})
+	if repos != nil {
+		repoArr, ok := repos.([]string)
+		if ok {
+			for _, r := range repoArr {
+				validRepos[r] = struct{}{}
+			}
+		}
+	}
+	for _, pod := range objects.Pods.Items {
+		diagnostics = append(diagnostics, pc.checkPod(validRepos, pod)...)
+	}
+
+	return diagnostics, nil
+}
+
+// checkPod checks if all pod images are well formed and
+// do not have any errors, or obvious mistakes.
+func (pc *containerImageCheck) checkPod(validRepos map[string]struct{}, pod corev1.Pod) []checks.Diagnostic {
+	var diagnostics []checks.Diagnostic
+	containers := append(pod.Spec.Containers, pod.Spec.InitContainers...)
```

Comment:
What about init containers? Similar to https://github.com/digitalocean/clusterlint/blob/master/checks/basic/fully_qualified_image.go#L56

Reply:
This does check both.